enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+
+## Motivation
+
+The current best practice for installing OpenShift 4 involves providing openshift-install with `root` cloud credentials powerful enough to mint additional fine grained credentials for each component in the cluster that needs them. The root credentials are stored in a secret in the cluster. This mechanism allows each component to have only the permissions it needs, and the required permissions are defined in the release image and automatically upgraded with the cluster itself.
+
+However some customers are unable or unwilling to do this and would prefer to disable automatic credential provisioning, and instead manage it themselves. It is possible to disable the cloud-credential-operator today by injecting a `ConfigMap` into the openshift-install manifests directory, along with several `Secrets` containing the cloud provider credentials the user manually configured. While this process exists today, it is not officially supported as we believe there are situations that could arise during upgrade where credentials have changed, and the upgrade will block and go into a failed state, with little visibility as to why or how it can be fixed.
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+Goals for users who have chosen to disable the cloud credential operator:
+
+  * Users can easily mint and inject their own cloud credential secrets at install time.
+  * Users will be prevented from upgrading *if* credentials have changed in an available release image update.
+  * Users can upgrade without failures when the new release image contains credential changes.
+  * Users have tooling at their disposal to aid them in detecting what credentials changes have occurred in a release image.
+  * Users have tooling at their disposal to aid them in updating their credentials with new permissions.
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to focus discussion
+and make progress.
+
+  * Providing a mechanism to disable the cloud credential operator, this mechanism already exists.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+This proposal requires the CCO to begin reporting Upgradable=False in a number of scenarios, unless an admin informs us they have taken action to reconcile new permissions and it is safe to upgrade.
+
+CLI tooling will be implemented to help users who wish to disable the CCO to perform their initial installations, update credentials prior to upgrade for release images available for update, and reconcile credentials against current CredentialsRequests in the cluster on an on-going basis. Users must take this manual step to clear the Upgradable=False condition if credentials have changed in any available update.
```

Comment:
> Users must take this manual step
what `this` manual step ?

Reply:
Run the reconcile --available-updates command below (as proposed), or do it manually and set the same flag to indicate it has been done. (i.e. add all available update versions to the reconciledAvailableUpdates list) Will clarify.